src/Nest/CommonAbstractions/SerializationBehavior/JsonFormatters/SourceFormatter.cs

```diff
@@ -30,23 +30,17 @@ public virtual void Serialize(ref JsonWriter writer, T value, IJsonFormatterReso
 			var settings = formatterResolver.GetConnectionSettings();
 
 			// avoid serialization to bytes when not using custom source serializer
-			if (ReferenceEquals(settings.SourceSerializer, settings.RequestResponseSerializer))
+			if (ReferenceEquals(settings.SourceSerializer, settings.RequestResponseSerializer)
+				|| settings.SourceSerializer is IInternalSerializerWithFormatter s && s.FormatterResolver != null)
```

Comment:
Would you be able to explain this change?

Reply:
I moved this out into its own method with `xmldocs` and used the check in `Deserialize` too.

Please let me know if this change make sense now and thank you for calling out to explain why and what!

Reply:
The bit that I'm unsure about is the additional OR check

```
settings.SourceSerializer is IInternalSerializerWithFormatter s && s.FormatterResolver != null)
```

is there a change elsewhere that now requires this?

Reply:
Addressed that in a follow up commits

https://github.com/elastic/elasticsearch-net/blob/f365b3070b75fecae14f176d7faf50010a87dfff/src/Nest/CommonAbstractions/SerializationBehavior/JsonFormatters/SourceFormatter.cs#L28-L36

Reply:
got it, thanks!